rc/lsp.kak

```diff
@@ -342,45 +342,32 @@ have_kakoune_feature_filtertext = ${kak_opt_lsp_have_kakoune_feature_filtertext}
 " | eval "${kak_opt_lsp_cmd} --request") > /dev/null 2>&1 < /dev/null & }
 }}
 
-declare-option -hidden str lsp_completion_selection ""
+declare-option -hidden str lsp_completion_inserted_range
 
-define-command -hidden lsp-completion-dismissed -docstring "Called when the completion pager is closed" %{
-    set-option window lsp_completion_selection %val{hook_param}
-    try %{
-        evaluate-commands -draft %{
-            select %opt{lsp_completion_selection}
-        }
-    } catch %{
-        # Completion was cancelled, nothing was inserted
-        # Clear these so the following lines do nothing.
-        # This is to avoid running them in the above try block, and silencing their errors.
-        remove-hooks window lsp-completion-dismissed
-        set-option window lsp_completions_selected_item -1
-    }
-    trigger-user-hook LSPCompletionDismissed
-    remove-hooks window lsp-completion-dismissed
-    set-option window lsp_completions_selected_item -1
-    set-option window lsp_completion_selection ""
+define-command -hidden lsp-completion-accepted -docstring "Called when a completion is accepted" %{
+    set-option window lsp_completion_inserted_range %val{hook_param}
```

Comment:
I didn’t know we could access `%val{hook_param}` in a command definition.

Reply:
yeah that's a bit strange